P4Plugin/Source/P4SpecCommand.cpp

```diff
@@ -10,49 +12,106 @@ class P4SpecCommand : public P4Command
 	{
 		ClearStatus();
 		m_Root.clear();
-		
+
+		Conn().Log().Info() << args[0] << "::Run()" << Endl;
+		std::string fallback_error = std::string();
+		if (args.size() > 1)
+			fallback_error = args[1];
+
 		const std::string cmd = std::string("client -o ") + Quote(task.GetP4Client());
-		
+
 		if (!task.CommandRun(cmd, this))
 		{
-			std::string errorMessage = GetStatusMessage();			
-			//This error is handled here instead of in P4LoginCommand as it only shows up when running the first non-login command (currently always spec)
+			//MFA: This error is handled here instead of in P4LoginCommand as it only shows up when running the first non-login command (currently always spec)
 			if (StatusContains(GetStatus(), "login2"))
 			{
 				ClearStatus();
-				GetStatus().insert(VCSStatusItem(VCSSEV_Error, "Your chosen Perforce server has multi-factor authentication enabled. To log in please use a visual client such as p4v or the \"p4 login\" or \"p4 login2\" CLI commands"));
+				std::string exePath = std::string(
+#if defined(_MACOS)
```

Comment:
and as this is also "platform specific", I would also move it perhaps just at the top of the file in order; use something like in P4InfoCommand

```
static const char* kClientName = "Client name: ";
static const char* kClientUnknown = "Client unknown.";
...
```

and use #elif 
and start with Windows always (since its >80% of the market/devs)

so here it would become;

```
#if defined(_WINDOWS)
static const char* kMFAExePath = "\"C:\\Program Files\\Perforce\\HelixMFA.exe\"";
#elif defined(_MACOS)
static const char* kMFAExePath = "/Applications/HelixMFA.app/Contents/MacOS/HelixMFA";
#elif defined(_LINUX)
static const char* kMFAExePath = "helixmfa";
#endif
```

Reply:
Change applied